sdk/cosmos/azure-cosmos/test/test_vector_policy.py

```diff
@@ -0,0 +1,252 @@
+
+    @classmethod
+    def setUpClass(cls):
+        if (cls.masterKey == '[YOUR_KEY_HERE]' or
+                cls.host == '[YOUR_ENDPOINT_HERE]'):
+            raise Exception(
+                "You must specify your Azure Cosmos account values for "
+                "'masterKey' and 'host' at the top of this class to run the "
+                "tests.")
+
+        cls.client = CosmosClient(cls.host, cls.masterKey)
+        cls.created_database = cls.client.get_database_client(test_config.TestConfig.TEST_DATABASE_ID)
+
+    def test_create_vector_embedding_container(self):
+        indexing_policy = {
+            "vectorIndexes": [
+                {"path": "/vector1", "type": "flat"},
+                {"path": "/vector2", "type": "quantizedFlat"},
+                {"path": "/vector3", "type": "diskANN"}
+            ]
+        }
+        vector_embedding_policy = {
+            "vectorEmbeddings": [
+                {
+                    "path": "/vector1",
+                    "dataType": "float32",
+                    "dimensions": 1000,
+                    "distanceFunction": "euclidean"
+                },
+                {
+                    "path": "/vector2",
+                    "dataType": "int8",
+                    "dimensions": 200,
+                    "distanceFunction": "dotproduct"
+                },
+                {
+                    "path": "/vector3",
+                    "dataType": "uint8",
+                    "dimensions": 400,
+                    "distanceFunction": "cosine"
+                }
+            ]
+        }
+        container_id = "vector_container" + str(uuid.uuid4())
+        created_container = self.created_database.create_container(
+            id=container_id,
+            partition_key=PartitionKey(path="/id"),
+            vector_embedding_policy=vector_embedding_policy,
+            indexing_policy=indexing_policy
+        )
+        properties = created_container.read()
+        assert properties["vectorEmbeddingPolicy"] == vector_embedding_policy
+        self.created_database.delete_container(container_id)
+
+    def test_fail_create_vector_indexing_policy(self):
+        vector_embedding_policy = {
+            "vectorEmbeddings": [
+                {
+                    "path": "/vector1",
+                    "dataType": "float32",
+                    "dimensions": 1000,
+                    "distanceFunction": "euclidean"
+                }]}
+
+        # Pass a vector indexing policy without embedding policy
+        indexing_policy = {
+            "vectorIndexes": [
+                {"path": "/vector1", "type": "flat"}]
+        }
+        try:
+            self.created_database.create_container(
+                id='vector_container',
+                partition_key=PartitionKey(path="/id"),
+                indexing_policy=indexing_policy
+            )
+            pytest.fail("Container creation should have failed for lack of embedding policy.")
```

Comment:
For my curiosity, is this check on the backend?
Shall we mention it in our SDK documentation somewhere? Specially in python API docs where we added the vector embedding policy.

Reply:
yep, all of the checks for the feature in the tests are done in the backend - will make a note of this too then

Reply:
@simorenoh , It seems reasonable to incorporate certain checks within the SDK client to reduce the number of backend calls. While I may not be familiar with all the dimensions involved, implementing checks on factors such as data type or distance policy appears feasible. Is there anything else I should consider?